src/app/pages/RewardPage/components/HistoryTable/TableBody/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { SkeletonRow } from 'app/components/Skeleton/SkeletonRow';
+import React, { useCallback } from 'react';
+import { TableRow } from '../TableRow/index';
+import { useTranslation } from 'react-i18next';
+import { translations } from '../../../../../../locales/i18n';
+import { weiTo4 } from 'utils/blockchain/math-helpers';
+
+export interface RewardEvent {
+  amount: string;
+  contract_address: string;
+  event: RewardEventType;
+  time: number;
+  user: string;
+  txHash: string;
+}
+
+export enum RewardEventType {
+  LIQUIDITY_MINING = 'liquidityMining',
+  VESTING = 'vesting',
+  STAKING = 'staking',
+  REFERRAL = 'referral',
+}
+
+interface ITableBodyProps {
+  items: RewardEvent[];
+  loading: boolean;
+}
+
+export const TableBody: React.FC<ITableBodyProps> = ({ items, loading }) => {
+  const { t } = useTranslation();
+
+  const getEvenetType = useCallback(
+    type => {
+      switch (type) {
+        case RewardEventType.LIQUIDITY_MINING:
+          return t(translations.rewardPage.historyTable.event.liquidityMining);
+        case RewardEventType.VESTING:
+          return t(translations.rewardPage.historyTable.event.vesting);
+        case RewardEventType.STAKING:
+          return t(translations.rewardPage.historyTable.event.staking);
+        case RewardEventType.REFERRAL:
+          return t(translations.rewardPage.historyTable.event.referral);
+        default:
+          return '';
+      }
+    },
+    [t],
+  );
+
+  return (
+    <tbody className="mt-5">
+      {items.map((item, index) => (
+        <TableRow
+          key={`${item.contract_address}/${index}`}
```

Comment:
This is not a good `key` because using `index` as a `key` is not recommended unless you have no other option: https://reactjs.org/docs/lists-and-keys.html#:~:text=We%20don't%20recommend%20using,cause%20issues%20with%20component%20state.&text=If%20you%20choose%20not%20to,to%20using%20indexes%20as%20keys. 

I'd suggest to use `txHash` as a `key` because it should be unique.